torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1097,6 +1104,7 @@ def create_wrapped_node(self, tx, args, kwargs, description):
             checkpoint,
             description,
             manually_set_subgraph_inputs=False,
+            should_flatten_outputs=True,
```

Comment:
Here's what we should do in WrapHigherOrderVariable.call_function:
- it calls speculate_subgraph with should_flatten_outputs=True
- it writes a call to `outputs = wrap(body_function, *args)`
- finally, it unflattens the output variables and returns that

In order to unwrap the output variables: perhaps we could construct a TorchVariable that is the pytree.tree_unflatten and invoke .call_function ?